cms/djangoapps/contentstore/views/course.py

```diff
@@ -1453,17 +1472,66 @@ def _get_usage_info(store, course, split_tests):
                 log.warning("Unit not found: %s", unit_location)
                 continue
 
-            unit_url = reverse_usage_url(
-                'container_handler',
-                course.location.course_key.make_usage_key(unit.location.block_type, unit.location.name)
+            usage_info = GroupConfiguration._get_usage_info(
+                course=course,
+                unit_parent=split_test,
+                unit=unit,
+                usage_info=usage_info,
+                group_id=split_test.user_partition_id,
+                scheme_name=UserPartition.RANDOM_SCHEME
             )
+        return usage_info
+
+    @staticmethod
+    def get_content_groups_usage_info(store, course):
+        """
+        Get usage information for content groups.
+        """
+        modules = store.get_items(course.id, settings={'group_access': {'$exists': True}})
```

Comment:
@explorerleslie @cahrens I did some performance testing for this query on a sandbox linked with prod clone. 

For HarvardX/ER22.1x/1T2014 this query takes around 0.02 seconds and for MITx/8.MReVx/2T2014 (which I think is the largest course), it takes around 1.15 seconds.